roundabout-api/src/main/java/at/fhv/itm3/s2/roundabout/api/entity/IStreetSection.java

```diff
@@ -67,33 +96,12 @@
     boolean isEnoughSpace(double length);
 
     /**
-     * Moves the first car in the section to the next section.
+     * Moves the first car from the current section to the next section.
+     * In background removes the first car (if there is one) from the queue and puts it into the
+     * queue of the next {@link IStreetSection} present in car route.
      */
     void moveFirstCarToNextSection();
 
-    /**
-     * Adds a new car to the street section.
-     *
-     * @param car The car to add.
-     */
-    void addCar(ICar car);
-
-    /**
-     * Gets first car in Section.
-     *
-     * @return first car in section
-     */
-    ICar getFirstCar();
-
-    /**
-     * Gets last car in Section.
-     *
-     * @return last car in section
-     */
-    ICar getLastCar();
-
-    /**
-     * removes the first car of the queue and returns the first Car
-     */
-    ICar removeFirstCar();
+    @Deprecated // TODO consider removal i think this logic can be packed into addCar method, othervise consider rename to isCarAbleToEnter()
```

Comment:
my suggestion was to convert addCar to a boolean method so it will work just like a regular limited-size collection -> true  if car was added to queue, false if not. why not to pack it into a car.. well tricky, but actually the information required for decision to be made (car could enter or not) is mainly stored StreetSection itself.